### DISCUSSION TITLE
Export page tree view as JSON

### DISCUSSION THREAD

**Discussion by dunklesToast** [vercel/next.js#32595] [Ideas]
### Describe the feature you'd like to request

The idea is to export the page tree view generated on `next build` as JSON. This would open many possibilities regarding DX and anomaly detection. Next.js users could monitor trends in page count and bundle sizes and can see issues before they go on production. 

E.g.:

Shared bundle is 200kB larger? Might want to check out the new dependencies you've added.
Page count dropped by 10? Might want to check your refactoring again.

This feature will be optional and won't affect any production performance as it's all done in the build process.

### Describe the solution you'd like

I've implemented a basic example of my idea at https://github.com/dunklesToast/next.js/commit/0f34afada5d4c53fb096b326e2482cd66f39de5c. It basically saves the data which is used to render the tree view as a JSON file in the build output folder.

The feature can be enabled via `generatePageData` config toggle.

### Describe alternatives you've considered

I've began writing a parser for the treeview but it is not as reliable as a JSON file and also does not provide byte-accurate file sizes.

**Comment by dunklesToast** [comment]
@balazsorban44 when can I expect feedback for this idea? It's over a month now and nothing has happened except that my [issue was converted to an discussion](https://github.com/vercel/next.js/issues/32593)

**Comment by cungminh2710** [comment]
Hi @balazsorban44, do you have any update/feedback for this idea? It's crucial to detect anomalies like huge bundle size, etc.
